docs/reference/query_proxy.rst

```diff
@@ -10,6 +10,7 @@ The ``ProxyQuery`` object is used to add missing features from the original `Doc
 * ``execute`` method - no need to call the ``getQuery()`` method,
 * add sort by and sort order options,
 * add preselect id query on left join query, so a limit query will be only applied on the left statement and not on the full select statement. This simulates the original Doctrine 1 behavior.
+* By default Sonata will use the ``DISTINCT`` SQL keyword, because ProxyQuery going to fetch identifiers. Sonata cannot detect whether or not you need ``DISTINCT``, but lets you remove that keyword in case it causes performance issues. To do so, simply call ``setDistinct(false)``.
```

Comment:
Please comply with the sf docs standards:

> [Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);](http://symfony.com/doc/current/contributing/documentation/standards.html)

Reply:
```patch
- because ProxyQuery going to fetch identifiers
+ because ``ProxyQuery`` is going to fetch identifiers
```

Reply:
I don't understand how this justifies the use of distinct, maybe someone else does?

Reply:
Is it maybe because when you do a join, you may have several rows for one id, and thus you won't be able to get the exact number of ids you want unless you use `DISTINCT` or `GROUP BY`?

Reply:
https://github.com/sonata-project/SonataAdminBundle/commit/c7f454ed86b0df5f636af90cff5cbac6cad45953#diff-21e8075cdf497941b782a6a3311751cb  - PR when DISTINCT was added.

Reply:
Ok I think I got it:
```
By default, Sonata will use the ``DISTINCT`` SQL keyword when fetching the identifiers of the entities that will be displayed in the listing, to avoid duplicates in some cases. Sonata cannot detect whether or not you need ``DISTINCT``, but lets you remove that keyword in case it causes performance issues and you are sure there will be no duplicates. To do so, simply call ``setDistinct(false)``.
```